### PR TITLE
check eth balance for fees

### PR DESCRIPTION
**Pull request by dragos-rebegea**

### CODE REVIEW

File: clients/ethereum/client.go [hunk 1]
```diff
@@ -46,6 +48,7 @@ type ArgsEthereumClient struct {
 }
 
 type client struct {
+	ethclient                 *ethclient.Client
```
Comment (iulianpascalau) [inline clients/ethereum/client.go:51]: please remove this. Instead, extend the ClientWrapper interface with the BalanceAt method.

File: factory/ethElrondBridgeComponents.go [hunk 2]
```diff
@@ -349,7 +350,13 @@ func (components *ethElrondBridgeComponents) createEthereumClient(args ArgsEther
 		return clients.ErrNilAddressConverter
 	}
 
+	ethClient, err := ethclient.Dial(ethereumConfigs.NetworkAddress)
```
Comment (iulianpascalau) [inline factory/ethElrondBridgeComponents.go:353]: not required

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: clients/gasManagement/disabled/disabledGasStation.go [hunk 3]
```diff
@@ -7,7 +7,7 @@ type DisabledGasStation struct{}
 
 // GetCurrentGasPrice returns nil,nil and will cause the gas price to be determined automatically
 func (dgs *DisabledGasStation) GetCurrentGasPrice() (*big.Int, error) {
-	return nil, nil
+	return big.NewInt(0), nil
```
Comment (iulianpascalau) [inline clients/gasManagement/disabled/disabledGasStation.go:10]: :+1:

File: testsCommon/interactors/blockchainClientStub.go [hunk 4]
```diff
@@ -41,6 +42,14 @@ func (bcs *BlockchainClientStub) ChainID(ctx context.Context) (*big.Int, error)
 	return big.NewInt(0), nil
 }
 
+func (bcs *BlockchainClientStub) BalanceAt(ctx context.Context, account common.Address, blockNumber *big.Int) (*big.Int, error) {
```
Comment (iulianpascalau) [inline testsCommon/interactors/blockchainClientStub.go:45]: missing comment

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by sstanculeanu — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**